From a84fac75f38de592e530a2f9f982d7aafec6c288 Mon Sep 17 00:00:00 2001
From: Steev Klimaszewski <steev@genesi-usa.com>
Date: Thu, 9 Jun 2011 19:28:48 -0500
Subject: [PATCH] [ARM] wire up sys_accept4() on ARM

sys_accept4() was added in kernel 2.6.28, but ARM was not updated
to include it.  The number and types of parameters is such that
no ARM-specific processing is needed, so wiring up sys_accept4()
just requires defining __NR_accept4 and adding a direct call in
the syscall entry table.
---
 arch/arm/include/asm/unistd.h | 1 +
 arch/arm/kernel/calls.S       | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/arch/arm/include/asm/unistd.h b/arch/arm/include/asm/unistd.h
index 8bac2fa..05a1435 100644
--- a/arch/arm/include/asm/unistd.h
+++ b/arch/arm/include/asm/unistd.h
@@ -391,6 +391,7 @@
 #define __NR_pwritev			(__NR_SYSCALL_BASE+362)
 #define __NR_rt_tgsigqueueinfo		(__NR_SYSCALL_BASE+363)
 #define __NR_perf_counter_open		(__NR_SYSCALL_BASE+364)
+#define __NR_accept4                    (__NR_SYSCALL_BASE+366)
 
 /*
  * The following SWIs are ARM private.
diff --git a/arch/arm/kernel/calls.S b/arch/arm/kernel/calls.S
index 9296c59..23ad31b 100644
--- a/arch/arm/kernel/calls.S
+++ b/arch/arm/kernel/calls.S
@@ -374,6 +374,8 @@
 		CALL(sys_pwritev)
 		CALL(sys_rt_tgsigqueueinfo)
 		CALL(sys_perf_counter_open)
+/* 365 */       CALL(sys_ni_syscall)        /* recvmmsg */
+                CALL(sys_accept4)
 #ifndef syscalls_counted
 .equ syscalls_padding, ((NR_syscalls + 3) & ~3) - NR_syscalls
 #define syscalls_counted
-- 
1.8.1.6
